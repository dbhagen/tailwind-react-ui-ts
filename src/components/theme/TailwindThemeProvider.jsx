import React from 'react'
import PropTypes from 'prop-types'
import merge from 'lodash.merge'

import defaultTheme from './defaultTheme'
import TailwindTheme from './TailwindTheme'

const TailwindThemeProvider = ({ theme, children }) => {
  const mergedTheme = merge({}, defaultTheme, theme)

  return (
    <TailwindTheme.Provider value={mergedTheme}>
      {children}
    </TailwindTheme.Provider>
  )
}

TailwindThemeProvider.propTypes = {
  theme: PropTypes.shape({}),
  children: PropTypes.node,
}

TailwindThemeProvider.defaultProps = {
  theme: {},
  children: undefined,
}

export default TailwindThemeProvider
