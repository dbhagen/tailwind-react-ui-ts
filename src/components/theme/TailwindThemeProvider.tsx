import React from 'react'
import merge from 'lodash.merge'

import defaultTheme from './defaultTheme'
import TailwindTheme from './TailwindTheme'

export type TailwindThemeProviderProps = {
  theme: {}
  children: React.ReactNode
}

const TailwindThemeProvider: React.FunctionComponent<TailwindThemeProviderProps> = ({ theme, children }) => {
  const mergedTheme = merge({}, defaultTheme, theme)

  return (
    <TailwindTheme.Provider value={mergedTheme}>
      {children}
    </TailwindTheme.Provider>
  )
}

export default TailwindThemeProvider
