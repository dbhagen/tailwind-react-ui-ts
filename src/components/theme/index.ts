export { default as defaultTheme } from './defaultTheme'
export { default as TailwindTheme } from './TailwindTheme'
export { default as TailwindThemeProvider } from './TailwindThemeProvider'
export { default as withTheme } from './withTheme'
export { default as useThemeValue } from './useThemeValue'
