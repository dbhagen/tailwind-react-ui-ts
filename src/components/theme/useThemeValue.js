import { tailwindPropToClassName } from '../tailwind'

export default (prefix, value, userClassNames = '') => {
  if (userClassNames.includes(prefix)) {
    return false
  }

  return tailwindPropToClassName(prefix, value)
}
