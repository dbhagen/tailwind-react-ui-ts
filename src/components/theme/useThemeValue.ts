import { tailwindPropToClassName } from '../tailwind'

export default (prefix: string, value: string, userClassNames = '') => {
  if (userClassNames.includes(prefix)) {
    return false
  }

  return tailwindPropToClassName(prefix, value)
}
