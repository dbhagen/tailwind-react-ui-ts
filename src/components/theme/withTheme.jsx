import React from 'react'

import TailwindTheme from './TailwindTheme'

const withTheme = Component => {
  const WithTheme = props => (
    <TailwindTheme.Consumer>
      {theme => <Component {...props} theme={theme} />}
    </TailwindTheme.Consumer>
  )

  WithTheme.displayName = `WithTheme(${Component.displayName})`

  return WithTheme
}

export default withTheme
