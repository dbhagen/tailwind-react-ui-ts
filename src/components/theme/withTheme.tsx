import React from 'react'

import TailwindTheme from './TailwindTheme'

export type themeType = {
  prefix: string
  radius: string
  spacing: {
    zero: number
    sm: number
    md: number
    lg: number
    xl: number
  }
  container: {
    sm: string
    md: string
    lg: string
  }
  text: {
    size: {
      body: string[]
      title: string[]
    },
    family: {
      body: string
      subtitle: string
      title: string
    }
  }
  brandColors: {
    primary: string
    secondary: string
    success: string
    danger: string
    warning: string
    info: string
  }
  textColors: {
    body: string
    link: string
    linkDark: string
    emphasis: string
    on: {
      primary: string
      secondary: string
      success: string
      danger: string
      warning: string
      info: string
      dark: string
    }
  }
  surfaceColors: {
    default: string
    dark: string
    light: string
  }
  highlightOffset: number
  accentSize: number
  breakpoints: {
    sm: string
    md: string
    lg: string
    xl: string
  }
}

export type withThemeProps = {
  theme: themeType
}

const withTheme = <T extends withThemeProps>(WrappedComponent: React.ComponentType<T & withThemeProps>) => {
  const WithTheme = (props: T) =>
    <TailwindTheme.Consumer >
      {theme => < WrappedComponent {...props} theme={theme} />}
    </TailwindTheme.Consumer >


  WithTheme.displayName = `WithTheme(${WrappedComponent.displayName})`

  return WithTheme
}

export default withTheme
