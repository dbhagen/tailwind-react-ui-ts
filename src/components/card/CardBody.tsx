import React from 'react'
import PropTypes from 'prop-types'

import { withTheme } from '../theme'
import { Box } from '../primitives'

const CardBody = ({ theme, is, children, ...rest }) => (
  <Box is={is} p={theme.spacing.md} {...rest}>
    {children}
  </Box>
)

CardBody.propTypes = {
  theme: PropTypes.shape({}).isRequired,
  is: PropTypes.oneOfType([PropTypes.string, PropTypes.func, PropTypes.object]),
  children: PropTypes.node,
}

CardBody.defaultProps = {
  is: 'div',
  children: undefined,
}

export { CardBody as component }
export default withTheme(CardBody)
