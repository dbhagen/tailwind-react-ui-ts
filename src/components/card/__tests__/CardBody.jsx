import React from 'react'
import { shallow } from 'enzyme'

import { component as CardBody } from '../CardBody'
import { defaultTheme } from '../../theme'

const setup = (testProps = {}) => {
  const props = Object.assign({ theme: defaultTheme }, testProps)

  const wrapper = shallow(<CardBody {...props}>Hello World!</CardBody>)

  return {
    props,
    wrapper,
  }
}

describe('CardBody', () => {
  it('renders matching snapshot', () => {
    const { wrapper } = setup()

    expect(wrapper).toMatchSnapshot()
  })
})
