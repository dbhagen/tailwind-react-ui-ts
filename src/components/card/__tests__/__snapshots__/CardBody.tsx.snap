// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardBody renders matching snapshot 1`] = `
<Box
  inline={false}
  inlineBlock={false}
  is="div"
  p={4}
>
  Hello World!
</Box>
`;
